onert-micro/onert-micro/include/core/OMTensorData.h

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright (c) 2025 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef ONERT_MICRO_CORE_TENSOR_DATA_H
+#define ONERT_MICRO_CORE_TENSOR_DATA_H
+
+#include "OMQuantizationData.h"
+
+#include <map>
+#include <memory>
+#include <type_traits>
+
+namespace onert_micro
+{
+namespace core
+{
+
+// clang-format off
+
+// ------------------------------------------------------------------------------------------------
+
+template <typename T>
+class OMTensorData
+{
+  using QuantData = OMQuantizationData<T>;
+  using QuantDataPtr = std::unique_ptr<QuantData>;
+
+private:
+  T *_data = nullptr;
+  size_t _size = 0;
+  QuantDataPtr _quant = {};
+  std::map<size_t, float> _out_float_data = {};
+
+public:
+  OMTensorData(T *data, const circle::Tensor *tensor)
+    : _data(data)
+  {
+    assert(data != nullptr);
+    assert(tensor != nullptr);
+
+    _size = OMRuntimeShape(tensor).flatSize();
+
+    if (IsQuantized<T>)
+    {
+      _quant = std::make_unique<QuantData>(_data, tensor);
+    }
+  }
+
+public:
+  template <typename U = T>
+  constexpr static bool IsInt8 = std::is_same<std::decay_t<U>, int8_t>::value;
+
+  template <typename U = T>
+  constexpr static bool IsUInt8 = std::is_same<std::decay_t<U>, uint8_t>::value;
+
+  template <typename U = T>
+  constexpr static bool IsQuantized = IsInt8<U> || IsUInt8<U>;
+
+public:
+  template <typename U = T>
+  constexpr static bool IsInput = std::is_const<U>::value;
+
+  template <typename U = T>
+  constexpr static bool IsOutput = !IsInput<U>;
```

Comment:
(Optional) M.b. it would be good to use `ByValue/ByReference` ? To avoid confusion with `NN` terms.